include/vm_phys.h

```diff
@@ -0,0 +1,42 @@
+#ifndef _VM_PHYS_H_
+#define	_VM_PHYS_H_
+
+#include <queue.h>
+#include <common.h>
+
+#define VM_NFREEORDER 9
+#define VM_RESERVED 1
+#define PAGE_SIZE 4096
+
+typedef uintptr_t vm_addr_t;
+typedef uintptr_t vm_paddr_t;
+
+struct vm_page {
+    TAILQ_ENTRY(vm_page) listq;
+    vm_addr_t virt_addr;
+    size_t size;
+
+    TAILQ_ENTRY(vm_page) freeq;
+    vm_paddr_t phys_addr;
+    size_t order;
+    uint32_t flags;
+};
```

Comment:
The convention inside kernel is that every `struct X` has a `typedef struct X X_t`.